File: Subarrays.py
def generate_subarrays(A, ans, temp, index, n):
    if index == n:
        ans.append(temp[:])
        return

    generate_subarrays(A, ans, temp, index + 1, n)
    temp.append(A[index])
    generate_subarrays(A, ans, temp, index + 1, n)
    temp.pop()

n = int(input())
A = list(map(int, input().split()))
ans = []
temp = []

generate_subarrays(A, ans, temp, 0, n)

for sub in ans:
    print(*sub)

File: Subarrays.cpp
#include <bits/stdc++.h>
using namespace std;

void Subarrays(vector<int>&A, vector<vector<int>>&ans,vector<int>temp,int index, int n)
{
    if(index==n)
    {
        ans.push_back(temp);
        return;
    }
    
    Subarrays(A,ans,temp,index+1,n);
    temp.push_back(A[index]);
    Subarrays(A,ans,temp,index+1,n);
    temp.pop_back();
    
}

int main() {
	int n;
	cin>>n;
	vector<int>A(n);
	for (int i=0;i<n;i++)
	    cin>>A[i];
	 vector<vector<int>>ans;
	 vector<int>temp;
	 
	 Subarrays(A,ans,temp,0,n);
	 
	 for (int i=0;i<ans.size();i++)
	 {
	     for (int j =0;j<ans[i].size();j++)
	     {
	         cout<<ans[i][j]<<" ";
	     }
	     cout<<"\n";
	 }
    
    return 0;
}
